domain/bakerystorage/state/config.go

```diff
@@ -0,0 +1,142 @@
+		OffersThirdPartyPrivateKey:        &keyScanner{offersThirdPartyKey.Private.Key},
+		OffersThirdPartyPublicKey:         &keyScanner{offersThirdPartyKey.Public.Key},
+	}
+
+	initialiseConfigStmt, err := sqlair.Prepare("INSERT INTO bakery_config (*) VALUES ($Config.*)", Config{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, initialiseConfigStmt, cfg).Run()
+		if internaldatabase.IsErrConstraintUnique(err) {
+			return BakeryConfigAlreadyInitialised
+		}
+		return domain.CoerceError(err)
+	})
+	return errors.Trace(err)
+}
+
+func (st *ConfigState) GetLocalUsersKey(ctx context.Context) (*bakery.KeyPair, error) {
+	cfg, err := st.getConfig(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return &bakery.KeyPair{
+		Public:  bakery.PublicKey{Key: cfg.LocalUsersPublicKey.Key()},
+		Private: bakery.PrivateKey{Key: cfg.LocalUsersPrivateKey.Key()},
+	}, nil
+}
+
+func (st *ConfigState) GetLocalUsersThirdPartyKey(ctx context.Context) (*bakery.KeyPair, error) {
+	cfg, err := st.getConfig(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return &bakery.KeyPair{
+		Public:  bakery.PublicKey{Key: cfg.LocalUsersThirdPartyPublicKey.Key()},
+		Private: bakery.PrivateKey{Key: cfg.LocalUsersThirdPartyPrivateKey.Key()},
+	}, nil
+}
+
+func (st *ConfigState) GetExternalUsersThirdPartyKey(ctx context.Context) (*bakery.KeyPair, error) {
+	cfg, err := st.getConfig(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return &bakery.KeyPair{
+		Public:  bakery.PublicKey{Key: cfg.ExternalUsersThirdPartyPublicKey.Key()},
+		Private: bakery.PrivateKey{Key: cfg.ExternalUsersThirdPartyPrivateKey.Key()},
+	}, nil
+}
+
+func (st *ConfigState) GetOffersThirdPartyKey(ctx context.Context) (*bakery.KeyPair, error) {
+	cfg, err := st.getConfig(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return &bakery.KeyPair{
+		Public:  bakery.PublicKey{Key: cfg.OffersThirdPartyPublicKey.Key()},
+		Private: bakery.PrivateKey{Key: cfg.OffersThirdPartyPrivateKey.Key()},
+	}, nil
+}
+
+func (st *ConfigState) getConfig(ctx context.Context) (Config, error) {
+	db, err := st.DB()
+	if err != nil {
+		return Config{}, errors.Trace(err)
+	}
+
+	q := "SELECT &Config.* FROM bakery_config"
+	loadConfigStmt, err := sqlair.Prepare(q, Config{})
+	if err != nil {
+		return Config{}, errors.Trace(err)
+	}
+
+	var cfgs []Config
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, loadConfigStmt).GetAll(&cfgs)
+		return err
+	})
+	if errors.Is(err, sql.ErrNoRows) {
+		return Config{}, errors.NotYetAvailablef("bakery config not yet initialised")
+	}
+	if err != nil {
+		return Config{}, errors.Trace(domain.CoerceError(err))
+	}
+	if len(cfgs) > 1 {
+		return Config{}, errors.Errorf("programming error: multiple bakery config items found")
```

Comment:
I think its safe to just use a `Get` above and not bother with this error.

There should be things like UNIQUE constraints to stop multiple of the same thing getting entered if this could be a problem.

Reply:
Done